Sources/StreamChatUI/ChatMessageList/ChatChannelHeaderView.swift

```diff
@@ -0,0 +1,137 @@
+
+/// The view that displays channel information on the message list header
+open class _ChatChannelHeaderView<ExtraData: ExtraDataTypes>:
+    _View,
+    ThemeProvider,
+    _ChatChannelControllerDelegate {
+    /// Controller for observing data changes within the channel.
+    open var channelController: _ChatChannelController<ExtraData>?
+
+    /// The user id of the current logged in user.
+    open var currentUserId: UserId? {
+        channelController?.client.currentUserId
+    }
+
+    /// Timer used to update the online status of member in the channel.
+    open var timer: Timer?
+
+    /// A view that displays a title label and subtitle in a container stack view.
+    open private(set) lazy var titleContainerView: TitleContainerView = components
+        .titleContainerView.init()
+        .withoutAutoresizingMaskConstraints
+
+    override open func setUp() {
+        super.setUp()
+
+        channelController?.setDelegate(self)
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        embed(titleContainerView)
+    }
+
+    override open func updateContent() {
+        super.updateContent()
+
+        titleContainerView.content = (titleText, subtitleText)
+
+        makeTimer()
+    }
+
+    /// The title text used to render the title label. By default it is the channel name.
+    open var titleText: String? {
+        guard let channel = channelController?.channel else { return nil }
+        return components.channelNamer(channel, currentUserId)
+    }
+
+    /// The subtitle text used in the subtitle label. By default it shows member online status.
+    open var subtitleText: String? {
+        guard let channel = channelController?.channel else { return nil }
+        guard let currentUserId = self.currentUserId else { return nil }
+
+        if channel.isDirectMessageChannel {
+            guard let member = channel
+                .lastActiveMembers
+                .first(where: { $0.id != currentUserId })
+            else {
+                return nil
+            }
+
+            if member.isOnline {
+                return L10n.Message.Title.online
+            } else if let minutes = member.lastActiveAt
+                .flatMap({ DateComponentsFormatter.minutes.string(from: $0, to: Date()) }) {
+                return L10n.Message.Title.seeMinutesAgo(minutes)
+            } else {
+                return L10n.Message.Title.offline
+            }
+        }
+
+        return L10n.Message.Title.group(channel.memberCount, channel.watcherCount)
+    }
+
+    /// Create a timer if the channel is direct between two people. Will call update content
+    /// repeatedly every minute to update the online status of the members.
+    open func makeTimer() {
+        if timer == nil, channelController?.channel?.isDirectMessageChannel == true {
+            timer = Timer.scheduledTimer(withTimeInterval: 60, repeats: true) { [weak self] _ in
```

Comment:
perhaps make the withTimeInterval configurable (via subclassing) and do not create the timer when set to 0

Reply:
Good idea 👍

Reply:
✅